### ISSUE TITLE
RabbitMQ consumers bug procesessing

### ISSUE THREAD

**Issue by timur-gafarov-acadaca** [magento/magento2#37851]
### Preconditions and environment

- Magento version - 2.4.5

### Steps to reproduce

- Create a rabbit mq message queue and push message in json format. Have some of the value to qual to null.
- Create a rabbit mq consumer and define properties in the constructor but do not define getters or setters. Extend from data object class
- When consuming message error will be thrown because core skips processing of keys that are equal to null

### Expected result

Message is consumed and passed to the processor class

### Actual result

Queue processor gets and empty results and no message payload data is present.

### Additional information

The issue is in this class:
`vendor/magento/framework/Webapi/ServiceInputProcessor.php` line 237

![image](https://github.com/magento/magento2/assets/127979857/6c22ead2-5f3d-40be-a926-3a8c3143f43d)

The issue is that if payload has null values, php function `isset` will return false. While here it is intended to check only existence of the key in the array disregarding value itself. This function should be changed to `array_key_exists`. What happens here is when this key is skipped with null, Magento falls back to check if getter method exists for the key which does not exist and breaks. This is clearly unintended because everything in the code suggests that you should be allowed to just define parameters in the constructer and extend from DataObject without the need to define all getters and setters.

In short once the function is changed this should fix the issue.

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-November** [comment]
Hello @timur-gafarov-acadaca,

We have tried to reproduce the issue on 2.4-develop, but in our case the issue was not reproducible.
Below are the steps we followed to reproduce the issue:
* Create a rabbit mq message queue and push message in json format. Have some of the value to qual to null.
* Create a rabbit mq consumer and define properties in the constructor but do not define getters or setters. Extend from data object class

Please re-verify the issue on your end, if the issue exists, kindly elaborate the steps and provide any screenshots that might help us to dig deep into the issue.

Thanks.

**Comment by engcom-November** [comment]
Hello @timur-gafarov-acadaca,

As this issue has not been updated from a long time, we assume this issue is fixed, hence closing the issue.
Feel free to raise a new issue or reopen this if you need more assistance.

Thank you.
